sia-gpu-miner.c

```diff
@@ -70,19 +71,22 @@ void quitSignal(int unused) {
 	printf("\nCaught kill signal, quitting...\n");
 }
 
+inline uint64_t get_timestamp_in_us()
+{
+#if defined(__linux__) || defined(__APPLE__)
+    struct timeval now;
+    gettimeofday(&now, NULL);
+    return now.tv_sec * 1000000LL + now.tv_usec;
+#else
+    return clock()/ CLOCKS_PER_SEC * 1000000LL;
```

Comment:
seems like there are a lot of ways to get the current timestamp. [This post](http://stackoverflow.com/a/12480485) describes the differences between various functions. I also found [this gist](https://gist.github.com/jbenet/1087739) by jbenet which shows how to get the current time on OS X. Not sure what the situation is like for Windows.